ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeer.java

```diff
@@ -522,7 +531,19 @@ public boolean hasSupportForMessage(final int messageCode) {
 
   @Override
   public String toString() {
-    return String.format("Peer %s...", nodeId().toString().substring(0, 20));
+    return String.format(
+        "Peer %s... %s, validated? %s, disconnected? %s",
+        getShortNodeId(), reputation, isFullyValidated(), isDisconnected());
+  }
+
+  @NotNull
+  public String getShortNodeId() {
+    return nodeId().toString().substring(0, 20);
+  }
+
+  @Override
+  public int compareTo(final @NotNull EthPeer ethPeer) {
+    return this.reputation.compareTo(ethPeer.reputation);
```

Comment:
Suggestion: if the reputation is the same, the compare by head state and finally by node id, so the ordering is more consistent

Reply:
expanded on this